electrum/gui/qml/qewallet.py

```diff
@@ -667,12 +671,56 @@ def deleteInvoice(self, key: str):
 
     @pyqtSlot(str, result=bool)
     def verifyPassword(self, password):
+        if not password:
+            password = None
         try:
             self.wallet.storage.check_password(password)
             return True
         except InvalidPassword as e:
             return False
 
+    @pyqtSlot(result=bool)
+    def canHaveKeystoreEncryption(self):
+        return self.wallet.can_have_keystore_encryption()
+
+    @pyqtSlot(str, result=bool)
+    def verifyKeystorePassword(self, password):
+        if not password:
+            password = None
+
+        if self.wallet.has_keystore_encryption() and not password:
+            return False
+
+        try:
+            self.wallet.keystore.check_password(password)
```

Comment:
I don't think this always behaves correctly, e.g. for a multisig wallet that has multiple keystores. This only checks the first keystore. I would prefer if we could just use `wallet.check_password()` directly, which has this logic encapsulated. None of the other GUIs call keystore.check_password() AFAICT.

Reply:
agreed